src/formatting/items.rs

```diff
@@ -1647,7 +1648,30 @@ fn rewrite_type<R: Rewrite>(
         } else {
             result.push(' ');
         }
-        let lhs = format!("{}=", result);
+
+        let comment_span = context
+            .snippet_provider
+            .opt_span_before(span, "=")
+            .map(|op_lo| mk_sp(generics.where_clause.span.hi(), op_lo));
+
+        let lhs = match comment_span {
+            Some(comment_span)
+                if contains_comment(context.snippet_provider.span_to_snippet(comment_span)?) =>
+            {
+                let comment_shape = Shape::indented(indent, context.config)
+                    .block_left(context.config.tab_spaces())?;
+
+                combine_strs_with_missing_comments(
+                    context,
+                    result.trim_end(),
+                    "=",
+                    comment_span,
+                    comment_shape,
+                    true,
+                )?
```

Comment:
We will want to use the `comment_shape` here for comments, but in cases where there's a where clause we don't want to for the assignment operator and the rhs. That's because in cases where the item has a `where` clause followed by a block or assignment the block/assignment is not supposed to be indented.

https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/items.md#where-clauses